akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -643,7 +640,16 @@ object Concat {
   /**
    * Create a new `Concat`.
    */
-  def apply[T](inputPorts: Int = 2): Concat[T] = new Concat(inputPorts)
+  def apply[T](inputCount: Int = 2) = GraphDSL.create() { implicit builder ⇒
+    import GraphDSL.Implicits._
+    val concat = builder.add(new Concat[T](inputCount))
+    val ds = concat.inSeq.map { inlet ⇒
+      val detacher = builder.add(new Detacher[T])
```

Comment:
This is already done, it is called `def detacher[T]` now, which should be used here.